### ISSUE TITLE
`kubernetes_manifest` panic when planning and importing

### ISSUE THREAD

**Issue by lawliet89** [hashicorp/terraform-provider-kubernetes#1411]
### Terraform Version, Provider Version and Kubernetes Version

```
Terraform v1.0.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/kubernetes v2.5.0Terraform v1.0.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/kubernetes v2.5.0
```

GKE `v1.20.9-gke.1001`

### Affected Resource(s)

- `kubernetes_manifest`

### Terraform Configuration Files
```hcl
terraform {
  required_providers {
    kubernetes = {
      source  = "hashicorp/kubernetes"
      version = "~> 2.5.0"
    }
  }
  required_version = ">= 1.0"
}

provider "kubernetes" {
   config_path    = "~/.kube/config"
  config_context = "staging"

  experiments {
    manifest_resource = true
  }
}

resource "kubernetes_manifest" "frontend_config" {
  manifest = {
    apiVersion = "networking.gke.io/v1beta1"
    kind       = "FrontendConfig"

    metadata = {
      name      = "ambassador-external"
      namespace = "core"
      labels    = {
        "app.kubernetes.io/instance" = "ambassador-external"
        "app.kubernetes.io/managed-by" = "Terraform"
        "app.kubernetes.io/name" = "ambassador"
        "app.kubernetes.io/part-of" = "ambassador-external"
      }
    }

    spec = {
      sslPolicy = "modern"

      redirectToHttps = {
        enabled          = true
        responseCodeName = "MOVED_PERMANENTLY_DEFAULT"
      }
    }
  }
}

```

### Panic Output

There are two panic outputs. In an already existing state file, the following panic appears when planning

```
Stack trace from the terraform-provider-kubernetes_v2.5.0_x5 plugin:

panic: AttributeName("labels"): can't use tftypes.Object["app.kubernetes.io/instance":tftypes.String, "app.kubernetes.io/managed-by":tftypes.String, "app.kubernetes.io/name":tftypes.String, "app.kubernetes.io/part-of":tftypes.String] as tftypes.Map[tftypes.String]

goroutine 623 [running]:
github.com/hashicorp/terraform-plugin-go/tftypes.NewValue(...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tftypes/value.go:267
github.com/hashicorp/terraform-plugin-go/tftypes.transform(0xc000fe52f0, 0x269e310, 0xc00400df80, 0x2025b20, 0xc003f56810, 0xc0055b58a0, 0x0, 0x1, 0xc000fe5128, 0xc0055da810, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tftypes/walk.go:170 +0x1bd1
github.com/hashicorp/terraform-plugin-go/tftypes.transform(0xc000fe52d8, 0x269e310, 0xc0055dccf0, 0x2025b20, 0xc0055db650, 0xc0055b58a0, 0x1, 0x2, 0xc0055dccf0, 0x2025b20, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tftypes/walk.go:163 +0x13a5
github.com/hashicorp/terraform-plugin-go/tftypes.Transform(...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tftypes/walk.go:112
github.com/hashicorp/terraform-provider-kubernetes/manifest/provider.(*RawProviderServer).PlanResourceChange(0xc0005a8380, 0x2690388, 0xc0034b6a40, 0xc0005fd7c0, 0xc00124a4f8, 0x1, 0x30)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/provider/plan.go:385 +0x2825
github.com/hashicorp/terraform-plugin-mux.SchemaServer.PlanResourceChange(0xc0005a6960, 0xc0005a6990, 0xc001490700, 0x2, 0x2, 0xc0005a8280, 0x1, 0x2690388, 0xc0034b6a40, 0xc0005fd7c0, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-mux/schema_server.go:276 +0x98
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).PlanResourceChange(0xc001490720, 0x2690430, 0xc0034b6a40, 0xc00167a1c0, 0xc001490720, 0xc003e645d0, 0xc00158cba0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:315 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler(0x22032c0, 0xc001490720, 0x2690430, 0xc003e645d0, 0xc001032840, 0x0, 0x2690430, 0xc003e645d0, 0xc000db4000, 0xff5)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:362 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000232fc0, 0x26a7ff8, 0xc000603080, 0xc00165d0e0, 0xc0005a6a20, 0x350fae8, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:1217 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000232fc0, 0x26a7ff8, 0xc000603080, 0xc00165d0e0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:1540 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0005941d0, 0xc000232fc0, 0x26a7ff8, 0xc000603080, 0xc00165d0e0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:878 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:876 +0x1fd

Error: The terraform-provider-kubernetes_v2.5.0_x5 plugin crashed!
```

When I try to create a very simplified example above and import it, I get 

```
Stack trace from the terraform-provider-kubernetes_v2.5.0_x5 plugin:

panic: can't create a tftypes.Value of type tftypes.Object["computed_fields":tftypes.List[tftypes.String], "field_manager":tftypes.List[tftypes.Object["force_conflicts":tftypes.Bool, "name":tftypes.String]], "manifest":tftypes.DynamicPseudoType, "object":tftypes.DynamicPseudoType, "timeouts":tftypes.List[tftypes.Object["create":tftypes.String, "delete":tftypes.String, "update":tftypes.String]], "wait_for":tftypes.Object["fields":tftypes.Map[tftypes.String]]], required attribute "computed_fields" not set

goroutine 145 [running]:
github.com/hashicorp/terraform-plugin-go/tftypes.NewValue(...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tftypes/value.go:267
github.com/hashicorp/terraform-provider-kubernetes/manifest/provider.(*RawProviderServer).ImportResourceState(0xc000828300, 0x2690388, 0xc00135cb40, 0xc001a380a0, 0xc0014465f8, 0x40e101, 0x10)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/provider/import.go:152 +0x2e25
github.com/hashicorp/terraform-plugin-mux.SchemaServer.ImportResourceState(0xc000826960, 0xc000826990, 0xc001519280, 0x2, 0x2, 0xc000828200, 0x1, 0x2690388, 0xc00135cb40, 0xc001a380a0, ...)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-mux/schema_server.go:298 +0x98
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ImportResourceState(0xc0015192a0, 0x2690430, 0xc00135cb40, 0xc001a2c0a0, 0xc0015192a0, 0xc001a281b0, 0xc001550ba0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:349 +0xdc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler(0x22032c0, 0xc0015192a0, 0x2690430, 0xc001a281b0, 0xc001a34060, 0x0, 0x2690430, 0xc001a281b0, 0xc001a3c080, 0x77)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:398 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000260fc0, 0x26a7ff8, 0xc001601500, 0xc001a36000, 0xc000826a20, 0x350fb18, 0x0, 0x0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:1217 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000260fc0, 0x26a7ff8, 0xc001601500, 0xc001a36000, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:1540 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00097e340, 0xc000260fc0, 0x26a7ff8, 0xc001601500, 0xc001a36000)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:878 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:876 +0x1fd

Error: The terraform-provider-kubernetes_v2.5.0_x5 plugin crashed!
```

Creating an entirely new resource does not seem to cause an issue.

This is the resource YAML as it exists in Kubernetes.

```yaml
apiVersion: networking.gke.io/v1beta1
kind: FrontendConfig
metadata:
  creationTimestamp: "2021-04-27T10:00:56Z"
  generation: 1
  labels:
    app.kubernetes.io/instance: ambassador-external
    app.kubernetes.io/managed-by: Terraform
    app.kubernetes.io/name: ambassador
    app.kubernetes.io/part-of: ambassador-external
  name: ambassador-external
  namespace: core
  resourceVersion: "422856911"
  uid: 76a02d18-942c-444f-9adc-55b950621c68
spec:
  redirectToHttps:
    enabled: true
    responseCodeName: MOVED_PERMANENTLY_DEFAULT
  sslPolicy: modern
```

### Expected Behavior
What should have happened?

To not crash.

### Actual Behavior
What actually happened?

Crashed

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by alexsomesan** [comment]
@lawliet89 Thanks for reporting this. 
It's really interesting and I'd like to dig deeper and get to the bottom of it, but I cannot reproduce it. The manifest you shared always applies correctly for me. 

Would you be able to share the existing state on which you are applying this? It doesn't need to be the entire state file, just the fragment for this resource. That would really help move things forward since I cannot reproduce out of the box.

Here's my results when applying this:
```
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # kubernetes_manifest.frontendconfig_core_ambassador_external will be created
  + resource "kubernetes_manifest" "frontendconfig_core_ambassador_external" {
      + manifest = {
          + apiVersion = "networking.gke.io/v1beta1"
          + kind       = "FrontendConfig"
          + metadata   = {
              + labels    = {
                  + app.kubernetes.io/instance   = "ambassador-external"
                  + app.kubernetes.io/managed-by = "Terraform"
                  + app.kubernetes.io/name       = "ambassador"
                  + app.kubernetes.io/part-of    = "ambassador-external"
                }
              + name      = "ambassador-external"
              + namespace = "default"
            }
          + spec       = {
              + redirectToHttps = {
                  + enabled          = true
                  + responseCodeName = "MOVED_PERMANENTLY_DEFAULT"
                }
              + sslPolicy       = "modern"
            }
        }
      + object   = {
          + apiVersion = "networking.gke.io/v1beta1"
          + kind       = "FrontendConfig"
          + metadata   = {
              + labels    = (known after apply)
              + name      = "ambassador-external"
              + namespace = "default"
            }
          + spec       = {
              + redirectToHttps = {
                  + enabled          = true
                  + responseCodeName = "MOVED_PERMANENTLY_DEFAULT"
                }
              + sslPolicy       = "modern"
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
kubernetes_manifest.frontendconfig_core_ambassador_external: Creating...
kubernetes_manifest.frontendconfig_core_ambassador_external: Creation complete after 0s

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Terraform v1.0.6
on darwin_amd64
+ provider registry.terraform.io/hashicorp/kubernetes v2.5.0

kubectl version --short
Client Version: v1.21.3
Server Version: v1.20.9-gke.701
```

**Comment by lawliet89** [comment]
Hey @alexsomesan you might be able to reproduce one of the crashes if you create the resource in GKE using the YAML and then try to import it.

I can send you the entire state file and some source code if you think it's useful. Could I have an email address?

**Comment by alexsomesan** [comment]
@lawliet89 OK. Let me try out the workflow with importing it and see if that triggers it. I'll let you know if it worked or I need the state file.

**Comment by sarahhodne** [comment]
I think I'm hitting the same issue, or at least a similar one, based on the stracktrace (can split this into a separate issue if you think it's unrelated).

The stack trace I get is:

```
panic: AttributeName("versions"): can't use tftypes.Tuple[tftypes.Object["additionalPrinterColumns":tftypes.List[tftypes.Object["description":tftypes.String, "format":tftypes.String, "jsonPath":tftypes.String, "name":tftypes.String, "priority":tftypes.Number, "type":tftypes.String]], "deprecated":tftypes.Bool, "deprecationWarning":tftypes.String, "name":tftypes.String, "schema":tftypes.Object["openAPIV3Schema":tftypes.DynamicPseudoType], "served":tftypes.Bool, "storage":tftypes.Bool, "subresources":tftypes.Object["scale":tftypes.Object["labelSelectorPath":tftypes.String, "specReplicasPath":tftypes.String, "statusReplicasPath":tftypes.String], "status":tftypes.Map[tftypes.String]]], tftypes.Object["additionalPrinterColumns":tftypes.List[tftypes.Object["description":tftypes.String, "format":tftypes.String, "jsonPath":tftypes.String, "name":tftypes.String, "priority":tftypes.Number, "type":tftypes.String]], "deprecated":tftypes.Bool, "deprecationWarning":tftypes.String, "name":tftypes.String, "schema":tftypes.Object["openAPIV3Schema":tftypes.DynamicPseudoType], "served":tftypes.Bool, "storage":tftypes.Bool, "subresources":tftypes.Object["scale":tftypes.Object["labelSelectorPath":tftypes.String, "specReplicasPath":tftypes.String, "statusReplicasPath":tftypes.String], "status":tftypes.Map[tftypes.String]]]] as tftypes.Tuple[tftypes.Object["additionalPrinterColumns":tftypes.List[tftypes.Object["description":tftypes.String, "format":tftypes.String, "jsonPath":tftypes.String, "name":tftypes.String, "priority":tftypes.Number, "type":tftypes.String]], "deprecated":tftypes.Bool, "deprecationWarning":tftypes.String, "name":tftypes.String, "schema":tftypes.Object["openAPIV3Schema":tftypes.DynamicPseudoType], "served":tftypes.Bool, "storage":tftypes.Bool, "subresources":tftypes.Object["scale":tftypes.Object["labelSelectorPath":tftypes.String, "specReplicasPath":tftypes.String, "statusReplicasPath":tftypes.String], "status":tftypes.Map[tftypes.String]]]]

goroutine 389 [running]:
github.com/hashicorp/terraform-plugin-go/tftypes.NewValue(...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tftypes/value.go:267
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.DeepUnknown(0x269e310, 0xc0053597a0, 0x269e310, 0xc005c4a270, 0x2025b20, 0xc0058c0c60, 0xc0014cf8f0, 0x1, 0xc005c8eae0, 0x2025b20, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/scaffold.go:38 +0x2f0a
github.com/hashicorp/terraform-provider-kubernetes/manifest/morph.DeepUnknown(0x269e310, 0xc0058b8960, 0x269e310, 0xc005c81e60, 0x2025b20, 0xc0058b8bd0, 0xc0014cf098, 0x2, 0xc005c81e60, 0x2025b20, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/morph/scaffold.go:29 +0x578
github.com/hashicorp/terraform-provider-kubernetes/manifest/provider.(*RawProviderServer).PlanResourceChange(0xc00013a800, 0x2690388, 0xc00065c100, 0xc0012e8370, 0xc0011b5108, 0x1, 0x30)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/manifest/provider/plan.go:326 +0x212e
github.com/hashicorp/terraform-plugin-mux.SchemaServer.PlanResourceChange(0xc0006deab0, 0xc0006deb10, 0xc0019285c0, 0x2, 0x2, 0xc00013a300, 0x1, 0x2690388, 0xc00065c100, 0xc0012e8370, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-mux/schema_server.go:276 +0x98
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).PlanResourceChange(0xc0019285e0, 0x2690430, 0xc00065c100, 0xc000380070, 0xc0019285e0, 0xc001533080, 0xc0004d2ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:315 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler(0x22032c0, 0xc0019285e0, 0x2690430, 0xc001533080, 0xc000d28f60, 0x0, 0x2690430, 0xc001533080, 0xc00106a000, 0xc86d)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:362 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00026d180, 0x26a7ff8, 0xc000902d80, 0xc000b0e000, 0xc0006deba0, 0x350fae8, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:1217 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00026d180, 0x26a7ff8, 0xc000902d80, 0xc000b0e000, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:1540 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00061b080, 0xc00026d180, 0x26a7ff8, 0xc000902d80, 0xc000b0e000)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:878 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-kubernetes/vendor/google.golang.org/grpc/server.go:876 +0x1fd

Error: The terraform-provider-kubernetes_v2.5.0_x5 plugin crashed!
```

The resources are:

```HCL
resource "kubernetes_manifest" "crd_volumesnapshotclasses" {
  manifest = yamldecode(file("snapshot.storage.k8s.io_volumesnapshotclasses.yaml"))
}

resource "kubernetes_manifest" "crd_volumesnapshotcontents" {
  manifest = yamldecode(file("snapshot.storage.k8s.io_volumesnapshotcontents.yaml"))
}

resource "kubernetes_manifest" "crd_volumesnapshots" {
  manifest = yamldecode(file("snapshot.storage.k8s.io_volumesnapshots.yaml"))
}
```

With the three files coming from [here](https://github.com/kubernetes-csi/external-snapshotter/tree/73f3def762aac526216ae8a213f477c34e836b7a/client/config/crd), but with the top-level `status` key removed.

In my case, I'm not importing the resources, they are brand new, but it's oddly flaky. On my local laptop it seems to plan normally most of the times, I've gotten it to crash once out of probably around 20-30 plans.

In CI it seems to fail more often, though sometimes with different errors. I at one point got this error instead, for an existing CRD resource (nothing had changed in prod for it):

```
╷
│ Error: Failed to morph manifest to OAPI type
│
│   with kubernetes_manifest.private_container_repository_crd,
│   on infra-operators.tf line 113, in resource "kubernetes_manifest" "private_container_repository_crd":
│  113: resource "kubernetes_manifest" "private_container_repository_crd" {
│
│ AttributeName("spec"): [AttributeName("spec")] failed to morph object element into object element:
│ AttributeName("spec").AttributeName("versions"): [AttributeName("spec").AttributeName("versions")] failed to
│ morph object element into object element: AttributeName("spec").AttributeName("versions"):
│ [AttributeName("spec").AttributeName("versions")] failed to morph tuple value: incompatible tuples
╵
```

**Comment by curtbushko** [comment]
I too am encountering this error when trying to apply the crds for knative-gcp. The CRDs are new in my GKE cluster.

```
manifest" "customresourcedefinition_brokercells_internal_events_cloud_google_com":
   1: resource "kubernetes_manifest" "customresourcedefinition_brokercells_internal_events_cloud_google_com" {
AttributeName("spec"): [AttributeName("spec")] failed to morph object element
into object element: AttributeName("spec").AttributeName("versions"):
[AttributeName("spec").AttributeName("versions")] failed to morph object
element into object element: AttributeName("spec").AttributeName("versions"):
[AttributeName("spec").AttributeName("versions")] failed to morph tuple value:
incompatible tuples
```

The specific manifest is:

```
resource "kubernetes_manifest" "customresourcedefinition_brokercells_internal_events_cloud_google_com" {
  manifest = {
    "apiVersion" = "apiextensions.k8s.io/v1"
    "kind" = "CustomResourceDefinition"
    "metadata" = {
      "labels" = {
        "events.cloud.google.com/crd-install" = "true"
        "events.cloud.google.com/release" = "v0.23.0"
      }
      "name" = "brokercells.internal.events.cloud.google.com"
    }
    "spec" = {
      "conversion" = {
        "strategy" = "Webhook"
        "webhook" = {
          "clientConfig" = {
            "service" = {
              "name" = "webhook"
              "namespace" = "cloud-run-events"
            }
          }
          "conversionReviewVersions" = [
            "v1",
            "v1beta1",
          ]
        }
      }
      "group" = "internal.events.cloud.google.com"
      "names" = {
        "categories" = [
          "knative-internal",
        ]
        "kind" = "BrokerCell"
        "plural" = "brokercells"
        "singular" = "brokercell"
      }
      "preserveUnknownFields" = false
      "scope" = "Namespaced"
      "versions" = [
        {
          "additionalPrinterColumns" = [
            {
              "jsonPath" = ".status.conditions[?(@.type==\"Ready\")].status"
              "name" = "Ready"
              "type" = "string"
            },
            {
              "jsonPath" = ".status.conditions[?(@.type==\"Ready\")].reason"
              "name" = "Reason"
              "type" = "string"
            },
            {
              "jsonPath" = ".metadata.creationTimestamp"
              "name" = "Age"
              "type" = "date"
            },
          ]
          "name" = "v1alpha1"
          "schema" = {
            "openAPIV3Schema" = {
              "properties" = {
                "spec" = {
                  "properties" = {
                    "components" = {
                      "properties" = {
                        "fanout" = {
                          "properties" = {
                            "avgCPUUtilization" = {
                              "format" = "int64"
                              "type" = "integer"
                            }
                            "avgMemoryUsage" = {
                              "type" = "string"
                            }
                            "cpuLimit" = {
                              "type" = "string"
                            }
                            "cpuRequest" = {
                              "type" = "string"
                            }
                            "maxReplicas" = {
                              "format" = "int64"
                              "type" = "integer"
                            }
                            "memoryLimit" = {
                              "type" = "string"
                            }
                            "memoryRequest" = {
                              "type" = "string"
                            }
                            "minReplicas" = {
                              "format" = "int64"
                              "type" = "integer"
                            }
                          }
                          "type" = "object"
                        }
                        "ingress" = {
                          "properties" = {
                            "avgCPUUtilization" = {
                              "format" = "int64"
                              "type" = "integer"
                            }
                            "avgMemoryUsage" = {
                              "type" = "string"
                            }
                            "cpuLimit" = {
                              "type" = "string"
                            }
                            "cpuRequest" = {
                              "type" = "string"
                            }
                            "maxReplicas" = {
                              "format" = "int64"
                              "type" = "integer"
                            }
                            "memoryLimit" = {
                              "type" = "string"
                            }
                            "memoryRequest" = {
                              "type" = "string"
                            }
                            "minReplicas" = {
                              "format" = "int64"
                              "type" = "integer"
                            }
                          }
                          "type" = "object"
                        }
                        "retry" = {
                          "properties" = {
                            "avgCPUUtilization" = {
                              "format" = "int64"
                              "type" = "integer"
                            }
                            "avgMemoryUsage" = {
                              "type" = "string"
                            }
                            "cpuLimit" = {
                              "type" = "string"
                            }
                            "cpuRequest" = {
                              "type" = "string"
                            }
                            "maxReplicas" = {
                              "format" = "int64"
                              "type" = "integer"
                            }
                            "memoryLimit" = {
                              "type" = "string"
                            }
                            "memoryRequest" = {
                              "type" = "string"
                            }
                            "minReplicas" = {
                              "format" = "int64"
                              "type" = "integer"
                            }
                          }
                          "type" = "object"
                        }
                      }
                      "type" = "object"
                    }
                  }
                  "type" = "object"
                }
                "status" = {
                  "properties" = {
                    "conditions" = {
                      "items" = {
                        "properties" = {
                          "lastTransitionTime" = {
                            "type" = "string"
                          }
                          "message" = {
                            "type" = "string"
                          }
                          "reason" = {
                            "type" = "string"
                          }
                          "severity" = {
                            "type" = "string"
                          }
                          "status" = {
                            "type" = "string"
                          }
                          "type" = {
                            "type" = "string"
                          }
                        }
                        "required" = [
                          "type",
                          "status",
                        ]
                        "type" = "object"
                      }
                      "type" = "array"
                    }
                    "ingressTemplate" = {
                      "description" = <<-EOT
                      IngressTemplate contains a URI template as specified by RFC6570 to generate Broker ingress URIs. It may contain variables `name` and `namespace`.
                      
                      EOT
                      "type" = "string"
                    }
                    "observedGeneration" = {
                      "format" = "int64"
                      "type" = "integer"
                    }
                  }
                  "type" = "object"
                }
              }
              "type" = "object"
            }
          }
          "served" = true
          "storage" = true
          "subresources" = {
            "status" = {}
          }
        },
      ]
    }
  }
}
```

**Comment by wombat** [comment]
I got the same error message as @lawliet89 and so far have not been able to import any type of resource with `kubernetes_manifest`.

@alexsomesan Without being familiar with the codebase I had a look at the `import.go:152` incl preceding lines mentioned in the stacktrace and in comparison to `plan.go` and `apply.go` I couldn´t find any reference to `computed_fields` - my rough guess is that somehow this required attribute is not getting set during `import` which would fit the error message. Can you verify this?

**Comment by bpoland** [comment]
I'm also seeing this when trying to import an [ExternalSecret](https://github.com/external-secrets/kubernetes-external-secrets) resource

```
Stack trace from the terraform-provider-kubernetes_v2.5.0_x5 plugin:

panic: can't create a tftypes.Value of type tftypes.Object["computed_fields":tftypes.List[tftypes.String], "field_manager":tftypes.List[tftypes.Object["force_conflicts":tftypes.Bool, "name":tftypes.String]], "manifest":tftypes.DynamicPseudoType, "object":tftypes.DynamicPseudoType, "timeouts":tftypes.List[tftypes.Object["create":tftypes.String, "delete":tftypes.String, "update":tftypes.String]], "wait_for":tftypes.Object["fields":tftypes.Map[tftypes.String]]], required attribute "computed_fields" not set
```

**Comment by alexsomesan** [comment]
I can confirm this is an issue that sneaked in while rebasing multiple PRs right before v2.5.0 release.
Should be fixed by PR #1422
